core/crypto/commitments/commitments.go

```diff
@@ -53,26 +51,25 @@ var (
 // Committer saves cryptographic commitments.
 type Committer interface {
 	// Write saves a cryptographic commitment and associated data.
-	Write(ctx context.Context, commitment []byte, committed *tpb.Committed) error
+	Write(ctx context.Context, commitment, data, nonce []byte) error
 	// Read looks up a cryptograpic commitment and returns associated data.
-	Read(ctx context.Context, commitment []byte) (*tpb.Committed, error)
+	Read(ctx context.Context, commitment []byte) (data, nonce []byte, err error)
 }
 
 // Commit makes a cryptographic commitment under a specific userID to data.
-func Commit(userID, appID string, data []byte) ([]byte, *tpb.Committed, error) {
+func Commit(userID, appID string, data []byte) (commitment, nonce []byte, err error) {
```

Comment:
Despite being better than before (as it does not depend on protobuf), this function is still not gobind friendly, as it returns 3 outputs. To be "visible" from Java, functions need to return only one output (plus an optional second one which needs to be of type error).